src/main/java/com/bookingApp/bookingApp/service/ReservationService.java

```diff
@@ -0,0 +1,181 @@
+                                ReservationRepository reservationRepository, TicketRepository ticketRepository){
+        this.screeningRepository = screeningRepository;
+        this.seatScreeningRepository = seatScreeningRepository;
+        this.reservationRepository = reservationRepository;
+        this.ticketRepository = ticketRepository;
+    }
+    public List<ScreeningsDto> findAll(){
+        List<ScreeningsDto> screeningsDtos = new ArrayList<>();
+        List<Screening> screenings = screeningRepository.findAll();
+        screenings.forEach(screening -> {
+            ScreeningsDto s = new ScreeningsDto();
+            s.setScreeningId(screening.getId());
+            s.setDate(screening.getScreeningDate());
+            s.setTitle(screening.getMovie().getTitle());
+            s.setRoom(screening.getSeatScreenings().get(0).getSeat().getRoom().getRoomNumber());
+
+            screeningsDtos.add(s);
+        });
+
+        Collections.sort(screeningsDtos, new Comparator<ScreeningsDto>() {
+            @Override
+            public int compare(ScreeningsDto o1, ScreeningsDto o2) {
+                if(o1.getTitle().compareTo(o2.getTitle())==0){
+                    return o1.getDate().compareTo(o2.getDate());
+                }else {
+                    return o1.getTitle().compareTo(o2.getTitle());
+                }
+            }
+        });
+
+        return screeningsDtos;
+    }
+    @Override
+    public List<ScreeningsDto> findAll(LocalDateTime dateFrom, LocalDateTime dateTo) {
+        List<ScreeningsDto> screeningsDtos = new ArrayList<>();
+        List<Screening> screenings = screeningRepository.readScreeningsByScreeningDateBetween(dateFrom,dateTo);
+        screenings.forEach(screening -> {
+            ScreeningsDto s = new ScreeningsDto();
+            s.setScreeningId(screening.getId());
+            s.setDate(screening.getScreeningDate());
+            s.setTitle(screening.getMovie().getTitle());
+            s.setRoom(screening.getSeatScreenings().get(0).getSeat().getRoom().getRoomNumber());
+
+            screeningsDtos.add(s);
+        });
+
+        Collections.sort(screeningsDtos, new Comparator<ScreeningsDto>() {
+            @Override
+            public int compare(ScreeningsDto o1, ScreeningsDto o2) {
+                if(o1.getTitle().compareTo(o2.getTitle())==0){
+                    return o1.getDate().compareTo(o2.getDate());
+                }else {
+                    return o1.getTitle().compareTo(o2.getTitle());
+                }
+            }
+        });
+        return screeningsDtos;
+    }
+
+    @Override
+    public ScreeningInfoDto findById(Long id) {
+        ScreeningInfoDto s = new ScreeningInfoDto();
+        Screening screening = screeningRepository.findById(id)
+                .orElseThrow(()->new ScreeningNotFoundException(id));
+        List<SeatDto> seatDtos = new ArrayList<>();
+
+        s.setScreeningId(screening.getId());
```

Comment:
czy można jakos lepiej ustawiać te wszystkie pola zamiast listy setterów? jaki to może generować problem?